vllm/v1/worker/gpu_model_runner.py

```diff
@@ -783,7 +783,13 @@ def _update_states(self, scheduler_output: "SchedulerOutput") -> None:
                 ] = spec_token_ids
                 # NOTE(woosuk): `num_tokens` here may include spec tokens.
                 self.input_batch.num_tokens[req_index] += num_spec_tokens
-                self.input_batch.spec_token_ids[req_index] = spec_token_ids
+
+            # When speculative decoding is used with structured output,
```

Comment:
I don't think it makes sense to have this comment located here. It doesn't seem related to the assignment in any way. Would it make more sense to have it on line 775 next to the `.get()`?